File: 739-daily-temperatures/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {

        int n = temperatures.size();
        vector<int> ans(n, 0);
        stack<pair<int, int>> st;
        for (int i = n - 1; i >= 0; i--) {
            while (!st.empty() && st.top().first <= temperatures[i]) {
                st.pop();
            }
            if (!st.empty()) {
                int count = st.top().second - i;
                ans[i] = count;
            }
            st.push({temperatures[i], i});
        }
        return ans;
    }
};

File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        st = []
        ans = []
        n = len(temperatures)
        for i in range(n-1, -1, -1):
            while(len(st)!=0 and temperatures[st[-1]] <= temperatures[i]):
                st.pop()
            if(len(st) == 0):
                ans.append(0)
            else:
                index = st[-1]
                ans.append(index - i)
            st.append(i)
        ans.reverse()
        return ans
                
